### DISCUSSION TITLE
Is there a property to expose TextEditorEvents?

### DISCUSSION THREAD

**Discussion by hrai** [VsixCommunity/Community.VisualStudio.Toolkit#137] [Q&A]
I need to intercept line changed event that lives inside this structure 

`dte.Events.TextEditorEvents.LineChanged`

Is there an equivalent property that I can use?

**Comment by madskristensen** [comment]
There might be an event source for this we can use like we do for the other events

**Reply by madskristensen** [reply]
@gundermanc how do I get the text editor events without using the DTE?

**Reply by gundermanc** [reply]
DTE is pretty much a wrapper around the 'real' editor APIs. There's no 1:1 version of this, but you can use ITextView.TextBuffer.Changed event to listen for changes in a document attached to a text view: https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.text.itextbuffer.changed?view=visualstudiosdk-2019

Now the hard part: where to get the ITextView reference.

- [ITextViewCreationListener](https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.text.editor.itextviewcreationlistener?view=visualstudiosdk-2019) - you can listen for text views to be created.
- If you are operating in an existing MEF extension, such as an intellisense completion source, you should typically be handed an ITextView or ITextBuffer when called.
